src/Serialise/Cborg/Tutorial.hs

```diff
@@ -16,33 +16,239 @@ module Serialise.Cborg.Tutorial
     -- ** The CBOR format
     -- $cborformat
 
+    -- ** The 'Serialise' class
+    -- $serialise
+
     -- ** Encoding terms
     -- $encoding
 
     -- ** Decoding terms
     -- $decoding
+
+    -- ** Migrations
+    -- $migrations
   ) where
 
+import Serialise.Cborg
+
 {- $introduction
 
-Lorem ipsum...
+@cborg@ is a library for the serialisation of Haskell values.
+
+As in modern serialisation libraries, @cborg@ offers
+instance derivation via GHC's 'GHC.Generic' mechanism,
+
+> import Serialise.Cborg
+> import qualified Data.ByteString.Lazy as BSL
+>
+> data Animal = HoppingAnimal { animalName :: String, hoppingHeight :: Int }
+>             | WalkingAnimal { animalName :: String, walkingSpeed :: Int }
+>             deriving (Generic)
+>
+> instance Serialise Animal
+>
+> fredTheFrog :: Animal
+> fredTheFrog = HoppingAnimal "Fred" 4
+>
+> main = BSL.writeFile "hi" (serialise fredTheFrog)
+
+We can then later read Fred,
+
+> main = do
+>     fred <- deserialise <$> BSL.readFile "hi"
+>     print fred
 
 -}
 
 {- $cborformat
 
-Lorem ipsum...
+@cborg@ uses the Concise Binary Object Representation, CBOR
+(IETF RFC 7049), as its serialised representation. This encoding is efficient in
+both encoding\/decoding complexity as well as space, and is generally
+machine-independent.
+
+The CBOR data model resembles that of JSON, having arrays, key\/value maps,
+integers, floating point numbers, binary strings, and text. In addition,
+CBOR allows items to be /tagged/ with a number which describes the type of data
+that follows. This can be used both to identify which data constructor of a type
+an encoding represents, as well as representing different versions of the same
+constructor.
+
+*** A note on interoperability
+
+@cborg@ is intended primarily as a /serialisation/ library for
+Haskell values. That is, a means of stably storing Haskell values for later
+reading by @cborg@. While it uses the CBOR encoding format, the
+library is /not/ primarily aimed to facilitate serialisation and
+deserialisation across different CBOR implementations.
+
+If you want to use @cborg@ to serialise\/deserialise values
+for\/from another CBOR implementation (either in Haskell or another language),
+you should keep a few things in mind,
+
+1. The 'Serialise' instances for some "basic" Haskell types (e.g. 'Maybe',
+   'Data.ByteString.ByteString', tuples) don't carry a tag, in contrast to common
+   convention. This is an intentional design decision to minimize encoding size.
```

Comment:
I would a comment that people writing their own instances should aim to follow the advice in the rest of this tutorial, as the only types which have weird encodings are ones which are never going to change, and authors should avoid deviating from the conventions to avoid future problems.